src/quick-lane/views/QuickLaneDetail.tsx

```diff
@@ -0,0 +1,248 @@
+import { ErrorView } from '../../error/views';
+import { ItemVideoDescription } from '../../item/components';
+import { InteractiveTour, LoadingErrorLoadedComponent, LoadingInfo } from '../../shared/components';
+import { CustomError, isMobileWidth } from '../../shared/helpers';
+import { trackEvents } from '../../shared/services/event-logging-service';
+import { QuickLaneService, QuickLaneUrlObject } from '../quick-lane.service';
+
+import './QuickLaneDetail.scss';
+
+interface QuickLaneDetailProps extends DefaultSecureRouteProps<{ id: string }> {}
+
+const QuickLaneDetail: FunctionComponent<QuickLaneDetailProps> = ({
+	history,
+	location,
+	match,
+	user,
+	...rest
+}) => {
+	// State
+	const [quickLane, setQuickLane] = useState<QuickLaneUrlObject>();
+	const [loadingInfo, setLoadingInfo] = useState<LoadingInfo>({ state: 'loading' });
+
+	const [t] = useTranslation();
+
+	// Retrieve data from GraphQL
+	const fetchQuickLaneAndContent = useCallback(async () => {
+		try {
+			const quickLaneId = match.params.id;
+
+			const response = await QuickLaneService.fetchQuickLaneById(quickLaneId);
+
+			trackEvents(
+				{
+					object: String(response.id),
+					object_type: 'quick_lane',
+					message: `Gebruiker ${getProfileName(
+						user
+					)} heeft een quick lane pagina bekeken`,
+					action: 'view',
+				},
+				user
+			);
+
+			setQuickLane(response);
+		} catch (err) {
+			console.error(
+				new CustomError('Failed to fetch quick lane and content for detail page', err, {
+					user,
+					id: match.params.id,
+				})
+			);
+			setLoadingInfo({
+				state: 'error',
+				message: t(
+					'assignment/views/assignment-detail___het-laden-van-de-opdracht-is-mislukt'
+				),
+			});
+		}
+	}, [setQuickLane, setLoadingInfo, match.params.id, t, user, history]);
+
+	useEffect(() => {
+		if (PermissionService.hasPerm(user, PermissionName.VIEW_ASSIGNMENTS)) {
+			fetchQuickLaneAndContent();
+		} else {
+			setLoadingInfo({
+				state: 'error',
+				message: t(
+					'assignment/views/assignment-detail___je-hebt-geen-rechten-om-deze-opdracht-te-bekijken'
+				),
+				icon: 'lock',
+			});
+		}
+	}, [fetchQuickLaneAndContent, user, t]);
+
+	useEffect(() => {
+		if (quickLane) {
+			setLoadingInfo({
+				state: 'loaded',
+			});
+		}
+	}, [quickLane]);
+
+	// Render methods
+	const renderContent = () => {
+		if (!quickLane || !quickLane.content) {
+			return null;
+		}
+
+		const contentLabel = quickLane.content_label;
+		const contentLayout = quickLane.view_mode;
+
+		switch (contentLabel) {
+			case 'COLLECTIE':
+				return (
+					<FragmentList
+						collectionFragments={
+							(quickLane.content as Avo.Collection.Collection).collection_fragments
+						}
+						showDescription={contentLayout === AssignmentLayout.PlayerAndText}
+						linkToItems={false}
+						history={history}
+						location={location}
+						match={match}
+						user={user}
+						collection={quickLane.content as Avo.Collection.Collection}
+						{...rest}
```

Comment:
Altijd oppassen als je onderaan nog een spread toevoegt omdat die bovenstaande props kan overschrijven. Als dat hier niet het geval kan zijn mag je het zo laten, gwn een fyi.

Reply:
View is gebaseerd op `src/assignment/views/AssignmentDetail.tsx`, ik laat het voor de zekerheid staan aangezien type hinting wel wat nuttige props toont.